mtags/src/main/scala-3/scala/meta/internal/pc/completions/Completions.scala

```diff
@@ -81,7 +81,8 @@ class Completions(
       val generalExclude =
         isUninterestingSymbol(sym) ||
           !isNotLocalForwardReference(sym) ||
-          sym.isPackageObject
+          sym.isPackageObject ||
+          sym.name.endsWith(Cursor.value)
```

Comment:
Will this not make no symbols to be shown?

Reply:
The only symbol with `CURSOR` we can get is the one we are on currently (i.e. we get node `Ident(<name>CURSOR)`, I don't think we would ever need completions with the same thing we just typed (after typing `val foobar@@` completion `foobar`)

Reply:
Hmm.... so it seems that we are getting scope completions with what currently is being defined :thinking: Maybe we can filter it out since it has an error type?

The problem I see here is that you can't define `CURSOR` value anymore, which I think wouldn't be an issue before.

Reply:
Oh right, so maybe it will be better to not suggest completions with symbols defined at current cursor position 🤔

Reply:
That might a good solution, we should get sourcePos and if that encapsulates the current pos we shouldn't show that

Reply:
> The problem I see here is that you can't define CURSOR value anymore, which I think wouldn't be an issue before.

Unfortunately, I don't think that's exactly correct. Having a CURSOR value is most probably problematic for some completions, since we perform quite a lot of `.replace(Cursor.value, "")`.

Reply:
So we might be deleting CURSOR suffix even if it wasn’t added by us? Maybe we can replace only the last occurrence of `CURSOR` in name?

Also, how about changing the string from `CURSOR` to `_CURSOR_` like in scala 2?

Reply:
Maybe stripSuffix would be enough then instead?

Reply:
I think it should be better, what do you think @kasiaMarek?

Reply:
I think this will produce issues if your cursor is in a middle of a word. `.replace(Cursor.value, "")` is usually done on some entity found in the tree, so it's not guaranteed to end with `CURSOR`. To be fair I'm not sure if such cases are handled reasonably anyway.

Reply:
Hmm... might be good to document this with a test at least @jkciesluk ?

Reply:
That should be ok.